reactor-core/src/main/java/reactor/core/publisher/FluxName.java

```diff
@@ -69,6 +69,13 @@
 
 		Set<Tuple2<String, String>> tags = Collections.singleton(Tuples.of(tagName, tagValue));
 
+		return createOrAppend(source, tags);
+	}
+
+	@SuppressWarnings("unchecked")
+	static <T> Flux<T> createOrAppend(Flux<T> source, Set<Tuple2<String, String>> tags) {
```

Comment:
mmh looking at this I think there is what could be considered a bug in how tags with the same key are handled. due to the `Tuple2`, the tags are ALWAYS additive so one could end up with `[example, foo]` and `[example, bar]` tags when calling `.tag("example", "foo").tag("example", "bar")`...

ultimately, if it gets transformed to a `Micrometer` `Tags` instance the later will deduplicate and it could appear to work, but that all depends on the iteration order (which is unspecified for `HashSet` 😱)

at public API level, this should be a `Map<String, String>`. Internally too, and source keys should get overwritten by new user-provided keys. Only the `scan` would keep the `Set<Tuple2>` type.